# AIPI 510: Sourcing for Data Analytics
# Week 6 Interactive: Power Analysis
# Caleb McNeill

### Chosen Claim: HydroHype Water Bottle

Carrying our smart water bottle increases average daily water consumption by at least 500 mL compared to standard bottles.
Measure: mL of water consumed (continuous)

### Null and Alternative Hypotheses

Null Hypothesis (H0): Carrying the smart water bottle does not increase average daily water consumption by at least 500 mL compared to standard bottles.

Alternative Hypothesis (H1): Carrying the smart water bottle increases average daily water consumption by at least 500 mL compared to standard bottles.

### Statistical Test to Perform 

The water consumption data should reflect a normal distribution because samples are continuous variables within a reasonable range and assumed to have a large sample size that can apply the Central Limit Theorem.

I will perform a one-sided independent samples t-test to compare the average daily water consumption of two groups of participants: those using the smart water bottle and those using standard bottles, against the hypothesized increase of 500 mL. Because there is no reason to expect that the standard bottle group would consume more water than the smart bottle group, a one-sided test is appropriate.

### Effect Size from Literature

Because I am using an independent samples t-test, the effect size can be represented by Cohen's d due to measuring the difference between the means of the two groups (smart water bottle group versus standard bottle group). I was unable to find studies specific to measuring water intake, but I did manage to find studies using smart water bottles and measuring urine output for Kidney Stone patients. According to a study completed in 2022, the mean difference in urine volume output between patient groups using smart water bottles (1.37 L) and being given simple dietary instruction (0.79 L) was approximately 580 mL. The pooled standard deviation between references was approximately 500 mL. Therefore, the estimated Cohen's d is  (580 - 500) / 500 ≈ 0.16, which indicates a small effect size according to conventional benchmarks. 

### Power Analysis

Because the study does not exhibit attributes that would warrant extradinary use of significance levels or power, I am choosing to use 0.05 for the significance level and 0.8 for the desired power.

In [3]:
### Power Analysis

from statsmodels.stats.power import TTestIndPower

# Define parameters for power analysis
effect_size = 0.16  # Cohen's d
alpha = 0.05  # Significance level (one-sided)
power = 0.8  # Desired power

# Initialize the power analysis object
analysis = TTestIndPower()

# Calculate the required sample size per group
n = analysis.solve_power(effect_size=effect_size, alpha=alpha, power=power, alternative='larger')
print(f"Required sample size per group: {n:.0f}")

Required sample size per group: 484


### Power Analysis Results

From the power analysis calculation, the required sample size per group to detect a small effect size (Cohen's d = 0.16) with a significance level of 0.05 and desired power of 0.8 is approximately 484 participants per group or 968 participants in total for both groups.

### References

Thomas E. Stout, et al., A Randomized Trial Evaluating the Use of a Smart Water Bottle to Increase Fluid Intake in Stone Formers. Journal of Urology, 2022. https://pubmed.ncbi.nlm.nih.gov/35283036/

Necole M. Streeper, et al., Feasibility of mini sipIT Behavioral Intervention to Increase Urine Volume in Patients with Kidney Stones. Journal of Urology, 2023. https://pmc.ncbi.nlm.nih.gov/articles/PMC11166357/

Alana C Desai, et al., Prevention of Urinary Stones with Hydration: A Randomised Clinical Trial of an Adherence Intervention. Lancet, 2026. https://pmc.ncbi.nlm.nih.gov/articles/PMC13043023/

